web/modules/custom/dpl_login/src/Controller/DplLoginController.php

```diff
@@ -110,14 +110,42 @@ public function logout(): TrustedRedirectResponse|RedirectResponse {
    */
   public function login(Request $request): Response {
     $_SESSION['openid_connect_op'] = 'login';
+
+    $url = Url::fromRoute('dpl_login.login_handler');
+
     if ($current_path = $request->query->get('current-path')) {
-      $_SESSION['openid_connect_destination'] = $current_path;
+      $url->mergeOptions(['query' => ['current-path' => $current_path]]);
+
     }
+    $generated_url = $url->toString(TRUE);
+    $_SESSION['openid_connect_destination'] = $generated_url->getGeneratedUrl();
 
     $scopes = $this->claims->getScopes($this->client);
     return $this->client->authorize($scopes);
   }
 
+  /**
+   * Check if a user token has been set and decide redirection.
+   *
+   * Check if a user token has been set and either allow redirecting
+   * to the original path or redirect to frontpage.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   Symfony request object.
+   *
+   * @return \Drupal\Core\Routing\TrustedRedirectResponse|\Symfony\Component\HttpFoundation\RedirectResponse
```

Comment:
Vil den ikke altid returnere `TrustedRedirectResponse` ?
```suggestion
   * @return \Drupal\Core\Routing\TrustedRedirectResponse
```

Reply:
Jo du har ret. Har fjernet den nu og opdateret beskrivelsen.